lib/src/__tests__/DateRangePicker.test.tsx

```diff
@@ -31,4 +31,19 @@ describe('<DateRangePicker />', () => {
     await waitFor(() => screen.getByRole('tooltip'));
     expect(screen.getByRole('tooltip')).toBeInTheDocument();
   });
+
+  it('prop – `renderDay` should be called and render days', async () => {
+    render(
+      <DesktopDateRangePicker
+        open
+        renderInput={defaultRangeRenderInput}
+        onChange={jest.fn()}
+        renderDay={day => <div key={String(day)} data-mui-test="renderDayCalled" />}
```

Comment:
We try to use the default test id attribute name when possible.

```suggestion
        renderDay={day => <div key={String(day)} data-testid="renderDayCalled" />}
```

cc @eps1lon Shouldn't we replace all usage of `data-mui-test` for `data-testid` in the core components too?

Reply:
I had that in mind yes. I'd prefer to keep them out of the core components though. This way people aren't confused why they aren't available in prod builds.

Reply:
I'm not sure we will be able to remove all the data test attributes from the source of the components (only using them in the test). Cases where it seems to be our only option:

- Icons, there are aria-hidden,
- Modal backdrop, no labels, no roles.